test-utils/src/dev/clickhouse.rs

```diff
@@ -720,11 +824,11 @@ impl ClickHouseDataDir {
     }
 }
 
-impl Drop for ClickHouseInstance {
+impl Drop for ClickHouseProcess {
     fn drop(&mut self) {
         if self.child.is_some() || self.data_dir.is_some() {
             eprintln!(
-                "WARN: dropped ClickHouseInstance without cleaning it up first \
+                "WARN: dropped ClickHouse process without cleaning it up first \
                 (there may still be a child process running and a \
                 temporary directory leaked)"
             );
```

Comment:
Nit: It could be helpful to know exactly if there is a process still running and/or temporary directory leaked. How about changing the message to only mention the process if `self.child.is_some()` and only mention the directory if `self.data_dir.is_some()`? That way the error message could even show what that directory is, so the user can find it. WDYT?

Reply:
It is printed already, in either the log files for tests or in the output of `ch-dev` or other `omicron-dev`-esque tools, but sure we can add it to this message as well.

Reply:
Added in ffdcaa73a